cni/log/logger.go

```diff
@@ -0,0 +1,56 @@
+package log
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+type Config struct {
+	Level       zapcore.Level
+	LogPath     string
+	MaxSizeInMB int
+	MaxBackups  int
+	Name        string
+}
+
+var Logger *zap.Logger
+
+// Initializes a Zap logger and returns a cleanup function so logger can be cleaned up from caller
+func Initialize(cfg *Config, ctx context.Context) {
+	Logger = newFileLogger(cfg)
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				Logger.Sync()
+				return
+			}
+		}
+	}()
+}
+
+func InitializeMock() {
+	Logger = zap.NewNop()
+}
```

Comment:
mocks should not be in real code, move this to the `_test.go`

Reply:
For this one, I choose to implement InitializeMock in logger_mock.go instead, since I can not use logger_test (if this was your suggestion) since when testing other packages that invoke InitializeMock, logger_test.go would not be built